CONTRIBUTING.rst

```diff
@@ -38,6 +38,61 @@ General Guidelines
   from the cmd line to run the test suite).
 - Add yourself to doc/contributors.rst :)
 
+Authoring a Pull Request
+------------------------
+
+**Our Pull Request Policy is based on this** `Code Review Developer Guide <https://google.github.io/eng-practices/review>`_
+
+The expectation for any code author is to provide all the context needed in the space of a
+pull request for any engineer to feel equipped to review the code. Depending on the type of
+change, do your best to highlight important new functions or objects you’ve introduced in the
+code—think complex functions or new abstractions. Whilst it may seem like more work for you to
+adjust your pull request, the reality is your likelihood for getting review sooner shoots
+up.
+
+**Self Review Guidelines to follow**
+
+- If the PR is too large, split it if possible.
+    - Use 400 LoC (excluding test data and config changes) as a rule-of-thumb
+    - Moving and changing code should be in separate PRs or commits. Rationale: it is easier for reviewers to see what functionality has changed.
```

Comment:
I don't quite understand this sentence, can you explain?

Reply:
I'll go on a limb here and assume it's the semantics around "moving" vs. "changing?" 
* Moving: Taking large code blobs and transplanting them to another file. There's generally no (or very little) actual code changed other than a cut and paste. I could even extend this to include sweeping file/directory deletions. 

* Changing: Adding code changes (be that refactors or functionality additions/subtractions)

The reason to disambiguate the two is that when reviewing code, these two can muddy understanding and sometimes make it harder for reviewers to keep track of things.

Reply:
Went ahead and added this descriptive snippet into the CONTRIBUTING.rst.